spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Container.scala

```diff
@@ -218,4 +219,26 @@ abstract class Container[A <: Activity : ClassTag,
     super.checkDuplicate(record)
     if (!skipDuplicateCheck()) modules.foreach(_.checkDuplicate(record))
   }
+
+  override def setDnnEngine(loc : Long): this.type = {
```

Comment:
why do we need to setEngine? Can we init an engine and when build layers always use that engine?

Reply:
Do we have multiple engines on one server?